mavisp/core.py

```diff
@@ -98,13 +98,20 @@ def _dataset_table(self, include, exclude):
                     curators= ', '.join(
                     [ f"{curator} ({', '.join(metadata['curators'][curator]['affiliation'])})" for curator in metadata['curators'].keys() ]
                     )
+                    uniprot_ac = ""
+                    if metadata['uniprot_ac']:
```

Comment:
this causes a crash if `uniprot_ac` isn't in the file. I think we can solve this in general by catching a `KeyError` exception below, similar to what we do for `IOError` - this would allow us to catch all cases in which the key doesn't exist in the yaml file. We should have a try/block for every key.

Something similar to:

```
try:
    metadata = self._parse_metadata(system, mode)
except IOError ...

if data is None: # parsing failed
    self.log.error(...)
    curators = None
    ...
...

try:
     uniprot_ac = metadata['uniprot_ac']
except KeyError:
    self.log.error(...)
    uniprot_ac = None

try: 
    refuse_id = metadata['refseq_id']
except KeyError:
....
```

Here you should set values as `None` when their parsing fails (as we do with `curators` already). This is caught later on and a Critical error is raised (which means the rest of the entries are not even parsed).